biogeochem/EDCanopyStructureMod.F90

```diff
@@ -1493,167 +1514,166 @@ subroutine leaf_area_profile( currentSite , snow_depth_si, frac_sno_eff_si)
        ! It is remotely possible that in deserts we will not have any canopy
        ! area, ie not plants at all...
        ! ------------------------------------------------------------------------------
-       
+
        if (currentPatch%total_canopy_area > nearzero ) then
 
 
-       currentCohort => currentPatch%tallest
-       do while(associated(currentCohort)) 
+          currentCohort => currentPatch%tallest
+          do while(associated(currentCohort)) 
 
-          ft = currentCohort%pft
-          cl = currentCohort%canopy_layer
+             ft = currentCohort%pft
+             cl = currentCohort%canopy_layer
 
-          ! Calculate LAI of layers above
-          ! Note that the canopy_layer_lai is also calculated in this loop
-          ! but since we go top down in terms of plant size, we should be okay
+             ! Calculate LAI of layers above
+             ! Note that the canopy_layer_lai is also calculated in this loop
+             ! but since we go top down in terms of plant size, we should be okay
 
-          leaf_c          = currentCohort%prt%GetState(leaf_organ,all_carbon_elements)
+             leaf_c          = currentCohort%prt%GetState(leaf_organ,all_carbon_elements)
 
-          currentCohort%treelai = tree_lai(leaf_c, currentCohort%pft, currentCohort%c_area, &
-                                           currentCohort%n, currentCohort%canopy_layer,               &
-                                           currentPatch%canopy_layer_tlai,currentCohort%vcmax25top )    
+             currentCohort%treelai = tree_lai(leaf_c, currentCohort%pft, currentCohort%c_area, &
+                  currentCohort%n, currentCohort%canopy_layer,               &
+                  currentPatch%canopy_layer_tlai,currentCohort%vcmax25top )    
 
-          currentCohort%treesai = tree_sai(currentCohort%pft, currentCohort%dbh, currentCohort%canopy_trim, &
-                                           currentCohort%c_area, currentCohort%n, currentCohort%canopy_layer, &
-                                           currentPatch%canopy_layer_tlai, currentCohort%treelai , &
-                                           currentCohort%vcmax25top,4)  
+             currentCohort%treesai = tree_sai(currentCohort%pft, currentCohort%dbh, currentCohort%canopy_trim, &
```

Comment:
For SP model, should "treesai" be prescribed rather than being calculated from allometric function?  This will lead to very small sai value.